tests-e2e/cypress/integration/runs/retrospective_spec.js

```diff
@@ -49,16 +49,26 @@ describe('runs > retrospective', () => {
         cy.visit(`/playbooks/runs/${runId}/retrospective`);
 
         // # Start editing
-        cy.findByText('Edit').click();
+        cy.findByTestId('retro-report-text').click();
 
         // * Verify the provided template text is pre-filled
         cy.focused().should('include.text', 'This is a retrospective template.');
 
-        // # Change the retro text, save it, and publish
+        // # Change the retro text
         cy.focused().clear().type('Edited retrospective.');
-        cy.findByText('Save').click();
+
+        // # Save it by clicking outside the text area
+        cy.findByText('Report').click();
+
+        // # Publish
         cy.findByText('Publish').click({force: true});
 
+        // * Verify we're showing the publish retro confirmation modal
+        cy.get('#confirmModalLabel').contains('Publish retrospective');
```

Comment:
Here we may have the possibility to use the dialog role:
```suggestion
        cy.findByRole('dialog').contains('Publish retrospective');
```
I have not tested it, though.

Reply:
Yah, there's only one role in that modal (it's `document`, probably not specific enough) -- the webapp gives us this id, so we can just use it. I don't know what the benefit of using something less specific would be...?

Reply:
From what I've seen writing the tests, thinking this way has forced me to change the DOM when needed to make it more accessible and semantic. For example, in this case it would be good that the modal had the `dialog` role. But it's not something we need to do now in this PR.